### ISSUE TITLE
Channel ID not sent to webhook endpoint

### ISSUE THREAD

**Issue by RoggerTan** [RocketChat/Rocket.Chat#33219]
### Description:

I'm having a critical issue after updating Rocket.Chat from 6.11.2 to 6.12.0. All of my bots, which are relying on outgoing webhook, became unusable. After some minutes of analyzing, It was the missing "channel_id" that caused the problem. 

### Steps to reproduce:
1. Create a bot user account. 
2. Create an outgoing webhook that listens to the bot's DM channel. 
3. Send a direct message to the bot. 

To reproduce the bug, I believe you only need to create a webhook without endpoint server. Try to trigger it and check out its' history data. 

### Expected behavior:

The endpoint server works and replies with appropriate response. 

### Actual behavior:

An exception occured at endpoint side as it needs channel_id for logging, future messaging,...

### Server Setup Information:

- Version of Rocket.Chat Server: 6.12.0
- License Type: Community 
- Number of Users: approx. 380 (but less than 10 active sessions for most of the time) 
- Operating System: Ubuntu Server 20.04.6 LTS
- Deployment Method: tar 
- Number of Running Instances: Omitted due to team's policies. 
- DB Replicaset Oplog: Omitted due to team's policies. 
- NodeJS Version: v14.21.3 
- MongoDB Version: 4.4.27 
I will include omitted information if it's essential. 

### Client Setup Information

- Desktop App or Browser Version: Both 
- Operating System: Varied 

### Additional context

I tried 3 fixes: 
1. Use Restful API: I got the message ID and sent to [chat.getMessage API](https://developer.rocket.chat/apidocs/get-message). Strangely, the **channels** field was an empty array. 
2. Give the bot higher roles: I even set it as admin, but it didn't work. 
3. Downgrade to 6.11.2: This worked, so I prefer staying outdated for a while. 

### Relevant logs:

Sorry. I reversed the update before saving the log.

**Comment by xcpdq** [comment]
I confirm webhooks/notifications stopped working. This is a SERIOUS issue! All our bots stopped working which is critical to our projects.

**Comment by reetp** [comment]
Bit confused here.

> Create an outgoing webhook that listens to the bot's DM channel.

Ok, so what does it contain?

> It was the missing "channel_id" that caused the problem.

You've told us about your attempts at a fix but not fully described the problem.

Which API call were you using?

Wondering if this should always have been included but was a bug that has been closed. But we need to establish exactly what you are doing first.

> Number of Running Instances: Omitted due to team's policies.
> DB Replicaset Oplog: Omitted due to team's policies.

Not sure what is sensitive about either of these.
Number of instances so we know what sort of setup you are running.
Replicaset is to make sure are running one.

> Downgrade to 6.11.2: This worked

Might have worked.

This is not recommended as there are usually database updates between versions that are not always backwards compatible. Only guaranteed method is backup/restore.

Not saving logs means it will be difficult to reproduce and debug. Perhaps you can run a  test rig to get some data?

And lastly, you should never upgrade to the latest release in production without testing thoroughly first. Personally I wait for a .1 release before looking at moving.....

**Comment by ugzv** [comment]
> I confirm webhooks/notifications stopped working. This is a SERIOUS issue! All our bots stopped working which is critical to our projects.

The same for us. I hope this gets resolved quickly.

**Comment by reetp** [comment]
Unless you tell us exactly what calls are made a "MeToo" means we can't help you.

Don't paste long logs here - add them.as attachments please.

**Comment by RoggerTan** [comment]
Here's the screenshot of outgoing data of both versions (6.11.2 & 6.12.0). The missing "channel_id" can be easily noticed. 
![image](https://github.com/user-attachments/assets/ee81cd53-80c2-432a-b793-c29cd93faf0d)
This is related to webhook, not Restful API. However, while trying the 1st fix, I found out that the [chat.getMessage](https://developer.rocket.chat/apidocs/get-message) API response didn't include any channel data. The `channels` field was a completely empty array: 
```
{
    "message": {
        "_id": "C6x7ya5zgQxgwMGjZ",
        "rid": "65898bd97791a26a02d56945",
        "msg": "@advisor info Tân",
        "ts": "2024-09-06T03: 58: 29.910Z",
        "u": {
            "_id": "Tun33BZJ5sDzrpfZM",
            "username": "15110123",
            "name": "Tester"
        },
        "_updatedAt": "2024-09-06T03: 58: 30.002Z",
        "urls": [],
        "mentions": [
            {
                "_id": "vJ4mfymmfTRbDt44Z",
                "username": "advisor",
                "name": "Advisor",
                "type": "user"
            }
        ],
        "channels": [],
        "md": [
            {
                "type": "PARAGRAPH",
                "value": [
                    {
                        "type": "MENTION_USER",
                        "value": {
                            "type": "PLAIN_TEXT",
                            "value": "advisor"
                        }
                    },
                    {
                        "type": "PLAIN_TEXT",
                        "value": " info Tân"
                    }
                ]
            }
        ]
    },
    "success": true
}

```
Sorry if the information wasn't much helpful. Since we've decided to keep the old version, I hope someone else could provide more, such as suggested log files.

**Comment by reetp** [comment]
> This is related to webhook

Yes you said.

So what code is in your webhook? You must be making some sort of call to notice the channel_id is missing and presumably to make you look at chat.getMessage ?

Just trying to establish the exact triggers so someone can try and replicate it.

Tell us all the (non private) info you have in your webhook or a complete sample so it can be tested.

**Comment by RoggerTan** [comment]
> So what code is in your webhook?

Oh just a simple Python Flask server. I named it "advisor bot" which has some basic features, such as member contact lookup and reminder. For instance, when I send: 
```
remind 10:00 today Team meeting
```
The advisor will receive the message along with other data provided by webhook feature. With some helps of string processing and a worker, When the time comes, advisor will send "Team meeting" to the channel that it received the task earlier. It's kinda like any reminder app/bot, the difference is I built it myself for team private use. I know, the bot name is kinda weird for its' jobs. 😄 

Anyway, the channel ID is required to know where (channel) to remind. Here's a sample data sent to incoming webhook endpoint: 
```
"text": "You have reminder(s) at 10:00 today: Team meeting",
'channel': "666a5c949b1e5bfd70d053ff",
"attachments": []
```

> Just trying to establish the exact triggers so someone can try and replicate it.

Understood. I'm quite busy, but I'll do my best. ^^

**Comment by reetp** [comment]
> I'm quite busy

We're all busy.....!

I believe this is being looked at internally.

**Comment by OrneyVortex** [comment]
if the issue is not solved yet?
can i start working on it?

**Comment by reetp** [comment]
> if the issue is not solved yet?
> can i start working on it?

If a bug is labelled "tasked" the internal teams are on it. There is a masive internal system you can't see.

But it's the weekend. Things take time.

By all means you can look at it but any PR is unlikely to be accepted.

**Comment by OrneyVortex** [comment]
ok sir got it!
i am new here!
I am trying for GSOC 25.
Will be contributing by all means as possible!
Thank you for your words

**Comment by reetp** [comment]
Checkout feature-requests repo, and have a look at older PRs here to see what happens & test.

**Comment by AndrewChau** [comment]
Same issue here after upgrading to 6.12

**Comment by RaijinRising** [comment]
Ditto

**Comment by luizjr** [comment]
> Same issue here after upgrading to 6.12

Experiencing the same issue here after update to 6.12.1

**Comment by reetp** [comment]
I think this is fixed here in 6.12.2

https://github.com/RocketChat/Rocket.Chat/pull/33306

**Comment by RoggerTan** [comment]
Can confirm the bug was solved in 6.13.0, so I'm closing the issue now.
